src/lib/IlmThread/IlmThreadPool.h

```diff
@@ -138,36 +139,37 @@ class ILMTHREAD_EXPORT ThreadPool
     // by threads as they become available, in FIFO order.
     //------------------------------------------------------------
 
-    void addTask (Task* task);
+    ILMTHREAD_EXPORT void addTask (Task* task);
     
 
     //-------------------------------------------
     // Access functions for the global threadpool
     //-------------------------------------------
     
-    static ThreadPool&	globalThreadPool ();
-    static void		addGlobalTask (Task* task);
+    ILMTHREAD_EXPORT static ThreadPool&	globalThreadPool ();
+    ILMTHREAD_EXPORT static void		addGlobalTask (Task* task);
 
-    struct Data;
+    struct ILMTHREAD_HIDDEN Data;
```

Comment:
Just curious, if the default is hidden, why do we have to mark this explicitly as HIDDEN?

Reply:
the outer class has been marked as public (default) such that it's typeinfo is exported, so I am forcing this one back to hidden. It may not be necessary on all platforms, but seemed safer.